app/src/main/java/org/wikipedia/feed/announcement/AnnouncementClient.java

```diff
@@ -24,72 +23,41 @@
 import java.util.Date;
 import java.util.List;
 
-import retrofit2.Call;
-import retrofit2.Response;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
 
 public class AnnouncementClient implements FeedClient {
     private static final String PLATFORM_CODE = "AndroidApp";
     private static final String PLATFORM_CODE_NEW = "AndroidAppV2";
 
-    @Nullable private Call<AnnouncementList> call;
+    private CompositeDisposable disposables = new CompositeDisposable();
 
     @Override
     public void request(@NonNull Context context, @NonNull WikiSite wiki, int age, @NonNull Callback cb) {
         cancel();
-        call = request(ServiceFactory.getRest(wiki));
-        call.enqueue(new CallbackAdapter(cb, true));
+        disposables.add(ServiceFactory.getRest(wiki).getAnnouncements()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(list -> {
+                    List<Card> cards = new ArrayList<>();
+                    if (list != null) {
```

Comment:
The null check is not necessary. If the observable is successful, the result is guaranteed to be nonnull. Otherwise it will fall through to the error block.
These five lines can be replaced with a single expression-lambda.

Reply:
Thanks, you're correct!

Reply:
@dbrant Done (in case you missed this).